src/yaml/mapping.cr

```diff
@@ -1,4 +1,63 @@
+# The `#yaml_mapping` macro defines how an object is mapped to YAML.
+#
+# Once `YAML` is required, `#yaml_mapping` macro is included by `Object`.
+# It takes `HashLiteral` as argument, in which attributes and types are defined.
+# Once defined, `Object#from_yaml` maps properties for the class based on the parsed 
+# YAML document.
+#
+# ```crystal
+# require "yaml"
+#
+# class Employee
+#   yaml_mapping({
+#     title: String,
+#     name: String
+#   })
+# end
+#
+# employee = Employee.from_yaml("title: Manager\nname: John")
+# employee.title #=> "Manager"
+# employee.name  #=> "John"
+#
+# employee.name = "Jenny"
+# employee.name #=> "Jenny"
+# ```
+#
+# Attributes not mapped with `#yaml_mapping` will not be defined as properties. 
+# Also, missing attributes raise error.
+#
+# ```crystal
+# employee = Employee.from_yaml("title: Manager\nname: John\nage: 30")
+# employee.age #=> undefined method 'age'
+#
+# employee = Employee.from_yaml("title: Manager")
+# #=> missing yaml attribute: name
+# ```
+# 
+# You can also define attributes for each mapped properties.
+#
+# ```crystal
+# class Employee
+#   yaml_mapping({
+#     title: String,
+#     name: {
+#       type: String,
+#       nilable: true,
+#       key: "firstname"
+#     }
+#   })
+# end
+# ```
+#
+# Available attributes: 
+#
+# * *type* (required) defines its type.
+# * *nilable* defines if a property can be `Nil`.
+# * *key* defines whick key to read from a YAML document.
+# * *converter* takes an alternate type for parsing and generation. A `#from_yaml`(`PullParser`) method is required in that class.
```

Comment:
I don't think the link is worth the odd formatting. Also mentioning that it should return an instance of what's given to `type` might be worth it.